retriever/engines/postgres.py

```diff
@@ -166,6 +166,53 @@ def insert_raster(self, path=None, srid=4326):
         if not path:
             path = Engine.format_data_dir(self)
 
+        if self.opts["bbox"] or self.script.tables[self.table.name].extent:
+
+            if self.opts["bbox"]:
+                xMin = self.opts["bbox"][0]
+                yMin = self.opts["bbox"][1]
+                xMax = self.opts["bbox"][2]
+                yMax = self.opts["bbox"][3]
+            else:
+                xMin = self.script.tables[self.table.name].extent["xMin"]
+                yMin = self.script.tables[self.table.name].extent["yMin"]
+                xMax = self.script.tables[self.table.name].extent["xMax"]
+                yMax = self.script.tables[self.table.name].extent["yMax"]
+
+            xMin = 0 if xMin < 0 else xMin
```

Comment:
@dassaniansh I will need some explanation  on this function from Line 182  to 200